### ISSUE TITLE
Swift 4 support

### ISSUE THREAD

**Issue by 3zcurdia** [onevcat/Kingfisher#781]
### Check List

Thanks for considering to open an issue. Before you submit your issue, please confirm these boxes are checked.

- [x] I have read the [wiki page](https://github.com/onevcat/Kingfisher/wiki) and [cheat sheet](https://github.com/onevcat/Kingfisher/wiki/Cheat-Sheet), but there is no information I need.
- [x] I have searched in [existing issues](https://github.com/onevcat/Kingfisher/issues?utf8=✓&q=is%3Aissue), but did not find a same one.
- [x] I want to reoprt a problem instead of asking a question. It'd better to use [kingfisher tag in Stack Overflow](http://stackoverflow.com/questions/tagged/kingfisher) to ask a question.

### Issue Description

Add support for swift 4

#### What

A project that complies with swift 4 marks the following issues:
-  Pods/Kingfisher/Sources/AnimatedImageView.swift:382:19: Declarations in extensions cannot override yet
- Pods/Kingfisher/Sources/Filter.swift:74:40: 'cropping(to:)' has been renamed to 'cropped(to:)'
- Pods/Kingfisher/Sources/Filter.swift:82:9: Closure tuple parameter '(CGFloat, CGFloat, CGFloat, CGFloat)' does not support destructuring
- Pods/Kingfisher/Sources/Image.swift:626:39: Static var 'colorControl' expects a single parameter of type 'Filter.ColorElement' (aka '(CGFloat, CGFloat, CGFloat, CGFloat)')

#### Reproduce

Create a new project in Xcode 9.0 add Kingfisher via cocoapods and try to compile

#### Other Comment

It can work if the project has been compiled for swift 3, but if you clean the project and try to compile again the compilation fails

**Comment by 3zcurdia** [comment]
temporal solution from stackoverflow, append this to your Podfile
```ruby

  post_install do |installer|
      installer.pods_project.targets.each do |target|
          if target.name == 'Kingfisher'
              target.build_configurations.each do |config|
                  config.build_settings['SWIFT_VERSION'] = '3.2'
              end
          end
      end
  end

```

**Comment by onevcat** [comment]
Kingfisher 4 already supports Swift 4. To use the latest version, please upgrade to version 4.

And you also need to change the Swift version to 4.0 in the build setting in Kingfisher. Or try to upgrade your CocoaPods.

See https://github.com/onevcat/Kingfisher/issues/772 https://github.com/onevcat/Kingfisher/issues/777 https://github.com/onevcat/Kingfisher/issues/779

Please, make sure you have **searched in existing issues, but did find a same one**

**Comment by 3zcurdia** [comment]
Sorry I did not realized that my cocoapods was not updated and I thought `3.13.1` and I did not check the closed issues/PR Thanks anyway

**Comment by onevcat** [comment]
:] Never mind. Since it seems that CocoaPods decides not to use setting provided in our podspec (https://github.com/CocoaPods/CocoaPods/issues/6791), I think it should be better to provide a version supports both Swift 3.2 and Swift 4, to avoid further issues like this. I will create a new version soon.

Thanks.

**Comment by onevcat** [comment]
Version 4.1.0 was released with supporting for both 3.2 and 4.0. That should fix this kind of issues.
